test/test_statement.rb

```diff
@@ -256,6 +256,12 @@ def test_named_bind_not_found
       stmt.close
     end
 
+    def test_params
+      stmt = SQLite3::Statement.new(@db, "select  ?1, :foo, ?, $bar, @zed, ?250, @999")
+      assert_equal ["foo", "bar", "zed"], stmt.named_params
```

Comment:
Related to my previous comment, I think this test needs to be updated and expanded to include numeric names:

```suggestion
      stmt = SQLite3::Statement.new(@db, "select  ?1, :foo, ?, $bar, @zed, ?250, @999, :123, $777")
      assert_equal ["foo", "bar", "zed", "999", "123", "777"], stmt.named_params
```